src/shogun/preprocessor/StringPreprocessor.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Wuwei Lin
+ */
+
+#include <shogun/preprocessor/StringPreprocessor.h>
+
+namespace shogun
+{
+
+	template <>
```

Comment:
why move this into implementation ?

Reply:
the header is mixed with declaration and implementation, so i split into .h and .cpp as they are done in other template classes, e.g. DensePreprocessor